Packages/vcs/Lib/VTKPlots.py

```diff
@@ -131,47 +140,97 @@ def leftButtonPressEvent(self, obj, event):
                 continue
             t = vcs.elements["template"][d.template]
             gm = vcs.elements[d.g_type][d.g_name]
-            if gm.projection != "linear":
-                return
-            if t.data.x1 <= x <= t.data.x2 and t.data.y1 <= y <= t.data.y2:
-                x1, x2, y1, y2 = vcs.utils.getworldcoordinates(gm,
-                                                               d.array[0].getAxis(-1),
-                                                               d.array[0].getAxis(-2))
-
-                X = (x - t.data.x1) / (t.data.x2 - t.data.x1) * (x2 - x1) + x1
-                Y = (y - t.data.y1) / (t.data.y2 - t.data.y1) * (y2 - y1) + y1
-
-                # Ok we now have the X/Y values we need to figure out the
-                # indices
-                try:
-                    I = d.array[0].getAxis(-1).mapInterval((X, X, 'cob'))[0]
+            # for non-linear projection or for meshfill. Meshfill is wrapped at
+            # VTK level, so vcs calculations do not work.
+            if gm.projection != "linear" or gm.g_name == 'Gfm':
+                selector = vtk.vtkHardwareSelector()
+                datasetRenderer = d.backend['dataset_renderer']
+                continentsRenderer = d.backend.get('continents_renderer')
+                dataset = d.backend['vtk_backend_grid']
+                if (datasetRenderer and dataset):
+                    selector.SetRenderer(datasetRenderer)
+                    selector.SetArea(xy[0], xy[1], xy[0], xy[1])
+                    selector.SetFieldAssociation(vtk.vtkDataObject.FIELD_ASSOCIATION_CELLS)
+                    # We want to be able see information behind continents
```

Comment:
That is correct. For point data, the user has to click fairly close the a point to see the value. I have to experiment with that, I am not sure.